guardian/tools-telemetry-service
# Add API Gateway and Lambda

_co-authored-by:_ @tjsilver

## What does this change?

Adds an API gateway and lambda to the CDK stack. The end goal is an endpoint which we can post telemetry events to, and a lambda service that pushes those events onto a Kinesis stream.

## How to test

Run `npm i; npm run test` in the `cdk` directory.

The tests should pass – it's just running the CDK to make sure there aren't any runtime errors.

Run `npm run synth`. Cloudformation should be successfully generated in the `cdk` folder.

## How can we measure success?

This code generates a stack that instantiates a correctly configured API gateway and lambda.

## Have we considered potential risks?

No risk, as nothing uses this service yet.

## Dev questions

 - ~Can we generate a certificate within this CDK, or must this happen in the AWS console?~ We can generate a cert in our region with this method, but it's preferable to have a global cert, which allows us to automatically provision a CDN. We've taken that approach for now.
 - ~Does this configuration look reasonable? (Neither author has much experience with CDK, or provisioning AWS services in general!)~ We'll find out on review 😁
